### PR TITLE
Fixes #2922 #2921 - Initial redesigned bug form integration

### LINKED ISSUE #2922: Integrate frontend from A/B testing experimental form with webcompat.com

### PR DESCRIPTION
**Pull request by johngian**
This PR fixes issue #2922 #2921 

## Proposed PR background

This is an attempt to start the integration of the frontend changes that @munteanu210 is currently working on. The rationale behind this PR after discussing it with @miketaylr is to:

* Make sure that the current webcompat.com functionality is not broken
*  Ensure that we don't break current tests
* Allow running new frontend designs in parallel with current webcompat.com state (A/B)
* Try to avoid nit-picking the new work since its experimental. The project changes suggested in this PR have a minor impact to the current webcompat.com codebase on purpose.
* Have a more sane way forward by opening issues for fixes to something actionable instead of building frontend code that piles up.

I understand that its a massive PR and shouldn't have been like that in the first place but I tried to cleanup the commits, make them more atomic, rebase on master. Some comments about the PR:

* For all the code (`bugform.js`, `form.html`) that was based on current functionality, I created a copy in a new commit and started committing changes on top of that so

  1. Its easier to isolate old and new functionality.
  2. We don't break current functionality
  3. Each easier to review diffs, rather than a huge blob

* All the new images/icons/svgs introduced are in separate commits
  * f724e4fbf21088b6715799b842bb29d4ef9d71ef
  * d0dbd3db4a87eb8fbdfe4094b82550dab512c971
* For easier review these are the only changes that might have impact to the current codebase
  * b34f136d4ef58dfdc91468cb2845d2207a5ec4d4
  * 1c3f77d804105227c91e0dd7e81ebe110e148cd1
 
I understand that the process is not ideal but I hope I made the review a little bit easier. Let me know what you think.

cc @miketaylr @karlcow 

---

- [x] I have read the [Pull Request + Code Style Guidelines](https://github.com/webcompat/webcompat.com/blob/master/docs/pr-coding-guidelines.md) thoroughly.

### CODE REVIEW

**Comment by karlcow** [comment]
> Try to avoid nit-picking the new work since its experimental. The project changes suggested in this PR have a minor impact to the current webcompat.com codebase on purpose.

Understood for this. 

This comes with a requirement though. We should not merge this branch ever in master. 
If the experiment is successful, we would then to have to recreate a PR with proper syntax, commits, and nit-picking 😉.

@johngian which type of review do you expect and what would be helpful for you?

File: webcompat/form.py [hunk 1]
```diff
@@ -111,12 +206,72 @@ class IssueForm(FlaskForm):
...
-def get_form(form_data):
+class PrefixedRadioField(RadioField):
+    """Prefix radio field label with an image."""
+    def __init__(self, *args, **kwargs):
+        prefixed_choices = kwargs.pop('choices')
+        template = '<div class={cls}><img src={src}/></div> {text}'
+        choices = []
+
+        for slug, img, text in prefixed_choices:
+            cls = 'icon-container'
```
Comment (karlcow) [inline webcompat/form.py]: This `cls` might trigger some bells for some syntax checkers, highlighters.
https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments

> Always use cls for the first argument to class methods.

This is not a reserved keyword, but this is used in different contexts.

File: webcompat/form.py [hunk 2]
```diff
@@ -111,12 +206,72 @@ class IssueForm(FlaskForm):
...
+            src = url_for('static', filename=filename)
+            label = Markup(template.format(src=src, cls=cls, text=text))
+            choice = (slug, label)
+            choices.append(choice)
+
+        kwargs['choices'] = choices
+        super(PrefixedRadioField, self).__init__(*args, **kwargs)
+
+
+class IssueFormRedesigned(IssueForm):
```
Comment (karlcow) [inline webcompat/form.py]: nit: Probably will deserve another title in the future. #tofixlater

File: webcompat/form.py [hunk 3]
```diff
@@ -111,12 +206,72 @@ class IssueForm(FlaskForm):
...
+    """Re-designed version of IssueForm:
```
Comment (karlcow) [inline webcompat/form.py]: nit: docstring doesn't follow our standard. #tofixlater

Though it's probably our fault to not enforce it through circleci check.

File: webcompat/form.py [hunk 4]
```diff
@@ -111,12 +206,72 @@ class IssueForm(FlaskForm):
...
+    Define form fields and validation for our bug reporting form.
+    """
+
+    steps = NEW_ISSUE_STEPS
+
+    browser = StringField(u'Browser', [Optional()])
+    device = StringField(u'Device', [Optional()])
+    os = StringField(u'Operating System', [Optional()])
```
Comment (karlcow) [inline webcompat/form.py]: `u` in `u'…'` should be dropped. Python 3 code.

File: webcompat/form.py [hunk 5]
```diff
@@ -111,12 +206,72 @@ class IssueForm(FlaskForm):
...
+        [InputRequired(message=radio_message)],
+        choices=video_bug_choices
+    )
+    browsers = PrefixedRadioField(
+        [InputRequired(message=radio_message)],
+        choices=browser_choices
+    )
+
+
+def get_form(form_data, form=IssueForm):
```
Comment (karlcow) [inline webcompat/form.py]: nit: In the future, this is probably a welcome improvement. #tofixlater
There could be a separate PR which basically give structure to the code for things without AB testing in a way that make it easier to configure. 

Thanks!

File: webcompat/helpers.py [hunk 6]
```diff
@@ -716,3 +716,7 @@ def ab_init(response):
             response.set_cookie(exp_id, var, max_age=max_age)
 
     return response
+
+
+# Register `ab_active` as jinja function
+app.jinja_env.globals.update(ab_active=ab_active)
```
Comment (karlcow) [inline webcompat/helpers.py]: I'm not a big fan of this hanging out in the bottom of a module function-less. That seems not at the right place. Was there a reason why it was not up there with cache_dict and other things.

File: webcompat/views.py [hunk 7]
```diff
@@ -175,7 +177,9 @@ def index():
         'rel': 'preload'
     })
     ua_header = request.headers.get('User-Agent')
+
```
Comment (karlcow) [inline webcompat/views.py]: That doesn't need to be here. 😁

File: webcompat/views.py [hunk 8]
```diff
@@ -175,7 +177,9 @@ def index():
...
     bug_form = get_form({'user_agent': ua_header})
+
```
Comment (karlcow) [inline webcompat/views.py]: That doesn't need to be here. 😁

File: config/__init__.py [hunk 9]
```diff
@@ -205,7 +205,8 @@ def update_status_config(milestones_content):
 AB_EXPERIMENTS = {
     'exp': {
         'variations': {
-            'form-v1': (0, 0),
+            'form-v1': (0, 100),
```
[the remark this answers is not in the document]
Comment (miketaylr) [reply config/__init__.py]: Does this mean we serve v2 (the new one?) 100% of the time?

File: webcompat/form.py [hunk 10]
```diff
@@ -15,11 +15,13 @@
 import re
 import urllib.parse
 
-from flask import g
+from flask import g, url_for
```
[the remark this answers is not in the document]
Comment (miketaylr) [reply webcompat/form.py]: This one seems simple to fix now. 😅

File: webcompat/form.py (hunk 2 above)
Comment (miketaylr) [reply webcompat/form.py]: maybe something like `StepByStepForm` or `FormWizard` i dunno

File: webcompat/helpers.py (hunk 6 above)
Comment (miketaylr) [reply webcompat/helpers.py]: @johngian can we follow the way we do this for `bust_cache`? It seems like it's equivalent, and would be nice to be consistent.

https://flask.palletsprojects.com/en/1.1.x/templating/#registering-filters

File: webcompat/templates/new-issue.html [hunk 11]
```diff
@@ -1,14 +1,33 @@
 {%- extends "layout.html" -%}
+
+{% block extracss %}
+{%- if ab_active('exp') == 'form-v2' %}
+<link href="{{ url_for('static', filename='css/src/issue-wizard.css') }}" type="text/css" rel="stylesheet">
```
Comment (miketaylr) [inline webcompat/templates/new-issue.html]: I like the name issue-wizard.

File: config/__init__.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (johngian) [reply config/__init__.py]: I just changed this one for convenience when starting testing in a live deployment. `form-v1` is current state, `form-v2` is the redesign which is served at 0%.

File: webcompat/form.py (hunk 5 above)
Comment (johngian) [reply webcompat/form.py]: Noted for the issues to be filled for future reference.

File: webcompat/helpers.py (hunk 6 above)
Comment (johngian) [reply webcompat/helpers.py]: Sure thing. I just followed the flask documentation and looked like a pretty common way to configure global functions.

**Comment by johngian** [comment]
Hey @miketaylr @karlcow I think I addressed most of the comments and I kept a couple of notes to open issues for future reference.

Regarding the last comment:

* I opened some issues to track the work already done. I am pretty sure we need to file more issues moving forward for the rest of the pending work
* I kept the `issue wizard` naming convention for the new files introduced

Let me know what you think

File: config/__init__.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (karlcow) [reply config/__init__.py]: thanks for the explanation

**Comment by karlcow** [comment]
I need to talk with @magsout to give a bit of context on this project.

**Pull request merged**